harness/propertyHelper.js

```diff
@@ -1,13 +1,17 @@
 
 function isConfigurable(obj, name) {
+    var descriptor = Object.getOwnPropertyDescriptor(obj, name);
     try {
         delete obj[name];
     } catch (e) {
         if (!(e instanceof TypeError)) {
             $ERROR("Expected TypeError, got " + e);
         }
     }
-    return !Object.prototype.hasOwnProperty.call(obj, name);
+    var result = !Object.prototype.hasOwnProperty.call(obj, name);
+    // Restore property (no-op if not deleted)
+    Object.defineProperty(obj, name, descriptor);
```

Comment:
Oh right, you can delete properties from non-extensible objects. Sorry for my confusion.

Well, would you object to adding a function which just does the test and doesn't do the delete? I'd like to use a factored-out verifyConfiguable, and adding the property back in my test feels like a layering violation (who would think that a function with that name would delete a property? It feels like depending on an implementation detail).